### ISSUE TITLE
Entity operations and datasetId fails

### ISSUE THREAD

**Issue by pierre-josselin** [Fiware/context.Orion-LD#1305]
**Orion-LD 1.0.0**
Context broker crashes.

**Orion-LD: 1.1.2**
207, Error treating attribute, sub-attribute must be a JSON object.

### Request

/ngsi-ld/v1/entityOperations/upsert

```json
[
    {
        "id": "urn:ngsi-ld:Vehicle:A135",
        "type": "Vehicle",
        "speed": {
            "type": "Property",
            "value": 55,
            "datasetId": "urn:ngsi-ld:Property:speedometerA4567-speed"
        }
    }
]
```

### Logs

```
fiware-orion     | time=Wednesday 18 Jan 11:49:45 2023.520Z | lvl=ERROR | corr=N/A | trans=N/A | from=N/A | srv=N/A | subsrv=N/A | comp=Orion | op=kjTreeToMetadata.cpp[82]:kjTreeToMetadata | msg=sub-attribute 'datasetId' of '' is not a JSON object
fiware-orion     | time=Wednesday 18 Jan 11:49:45 2023.521Z | lvl=ERROR | corr=N/A | trans=N/A | from=N/A | srv=N/A | subsrv=N/A | comp=Orion | op=kjTreeToUpdateContextRequest.cpp[139]:kjTreeToContextElementAttributes | msg=kjTreeToContextAttribute failed for attribute 'https://uri.etsi.org/ngsi-ld/default-context/speed': sub-attribute must be a JSON object
fiware-orion     | time=Wednesday 18 Jan 11:49:45 2023.522Z | lvl=WARN | corr=N/A | trans=N/A | from=N/A | srv=N/A | subsrv=N/A | comp=Orion | op=kjTreeToUpdateContextRequest.cpp[304]:kjTreeToUpdateContextRequest | msg=kjTreeToContextElementAttributes flags error 'Error treating attribute: sub-attribute must be a JSON object' for entity 'urn:ngsi-ld:Vehicle:A135'
```

**Comment by pierre-josselin** [comment]
Similar to #1246

**Comment by kzangeli** [comment]
Yeah, sorry about that.
datasetId isn't implemented in Orion-LD, except for a PoC that was made for, I think, two operations only.
It's also not especially high on the backlog for Orion-LD, as more interesting features, such as distributed operations or attribute grouping will be dealt with before.

If you truly need datasetId, I can only direct you to Scorpio, that already supports it.

**Comment by kzangeli** [comment]
That said, a proper error message should be returned, and definitely not a crash.
I'll look into that asap

**Comment by pierre-josselin** [comment]
Thank you for your reply. Please note that the crash occurs in 1.0.0.
In 1.1.2 there is no crash, but an error message.

**Comment by kzangeli** [comment]
Yeah, I saw, thanks.
Don't think I'll fix the 1.0.0, but, I'll make sure that all operations give a nice "501 Not Implemented", which is how the broker should respond to this.

Thanks for reporting!
